homeassistant/components/sensor/iperf3.py

```diff
@@ -0,0 +1,230 @@
+"""
+Support for Iperf3 network measurement tool.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.iperf3/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN, PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_MONITORED_CONDITIONS, STATE_UNKNOWN)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import slugify
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_change
+from homeassistant.helpers.restore_state import async_get_last_state
+import homeassistant.util.dt as dt_util
+
+REQUIREMENTS = ['iperf3==0.1.10']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_PROTOCOL = 'Protocol'
+ATTR_REMOTE_HOST = 'Remote Server'
+ATTR_REMOTE_PORT = 'Remote Port'
+ATTR_TEST_STATUS = 'Test Status'
+ATTR_VERSION = 'Version'
+
+CONF_ATTRIBUTION = 'Data retrieved using Iperf3'
+CONF_SECOND = 'second'
+CONF_MINUTE = 'minute'
+CONF_HOUR = 'hour'
+CONF_DAY = 'day'
+CONF_MANUAL = 'manual'
+CONF_DURATION = 'duration'
+CONF_SERVER = 'server'
+CONF_PORT = 'port'
+
+DEFAULT_DURATION = 10
+DEFAULT_PORT = 5201
+
+ICON = 'mdi:speedometer'
+
+SENSOR_TYPES = {
+    'download': ['Download', 'Mbit/s'],
+    'upload': ['Upload', 'Mbit/s'],
+}
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(list(SENSOR_TYPES))]),
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Required(CONF_SERVER): cv.string,
+    vol.Optional(CONF_DURATION, default=DEFAULT_DURATION): vol.Range(5, 10),
+    vol.Optional(CONF_SECOND, default=[0]):
+        vol.All(cv.ensure_list, [vol.All(vol.Coerce(int), vol.Range(0, 59))]),
+    vol.Optional(CONF_MINUTE, default=[0]):
+        vol.All(cv.ensure_list, [vol.All(vol.Coerce(int), vol.Range(0, 59))]),
+    vol.Optional(CONF_HOUR):
```

Comment:
Please remove this. For platforms Home Assistant already allows users to override a scan interval by passing `scan_interval: <duration>` to the platform config. This is defined in `PLATFORM_SCHEMA` which you extend. To set a default, define a `SCAN_INTERVAL = timedelta(minutes=30)`